unix/Makefile

```diff
@@ -108,6 +114,19 @@ SRC_MOD += modjni.c
 endif
 
 # source files
+ifeq ($(TARGET), freedos)
+# no moduselect for freedos
+SRC_C = \
+	main.c \
+	gccollect.c \
+	unix_mphal.c \
+	input.c \
+	file.c \
+	modos.c \
+	alloc.c \
+	coverage.c \
+	$(SRC_MOD)
+else
```

Comment:
If it's all about "no moduselect for freedos", then moduselect.c should be #if'ed for MICROPY_PY_USELECT (which needs to be added to mpconfigport*.h (not py/mpconfig.h)). That's the preference we have - doing stuff on C level instead of Makefile level if possible.

This should be a separate patch if you go for it.